carpool-map/js/map.js

```diff
@@ -1,151 +1,80 @@
-		if (count <= GRADUATED_POINT_COUNTS[i]){
-			return mult * r;
-		}
-	}
-	return mult * r;
-};
-
-function style(feature, color){
-	return [new ol.style.Style({
-		image: new ol.style.Circle({
-			radius: radius(feature.getCount()),
-			fill: new ol.style.Fill({
-				color: 'rgba(' + color.toString() + ',0.5)'
-			}),
-			stroke: new ol.style.Stroke({
-				color: 'rgba(' + color.toString() + ',1)',
-				width: 2
-			})
-		})
-	})];
-};
-
-view = new ol.View({center: [-8131982, 4994701], zoom: 11});
-
-map = new ol.Map({
-	target: $('#map').get(0),
-	layers: [new ol.layer.Tile({source: new ol.source.OSM()})],
-	view: view
-});
-
-riderSrc = new nyc.ol.source.Decorating(
-	{loader: new nyc.ol.source.CsvPointFeatureLoader({
-		url: RIDERS_URL + bust(),
-		projection: 'EPSG:4326',
-		xCol: LONGITUDE_COL,
-		yCol: LATITUDE_COL
-	})},
-	helpers,
-	{projection: 'EPSG:3857'}
-);
-
-riderLyr = new ol.layer.Vector({
-	source: riderSrc,
-	style: function(feature){
-		return style(feature, RIDER_COLOR);
-	}
-});
-
-driverSrc = new nyc.ol.source.Decorating(
-	{loader: new nyc.ol.source.CsvPointFeatureLoader({
-		url: DRIVERS_URL + bust(),
-		projection: 'EPSG:4326',
-		xCol: LONGITUDE_COL,
-		yCol: LATITUDE_COL
-	})},
-	helpers,
-	{projection: 'EPSG:3857'}
-);
-
-driverLyr = new ol.layer.Vector({
-	source: driverSrc,
-	style: function(feature){
-		return style(feature, DRIVER_COLOR);
-	}
-});
-
-map.addLayer(riderLyr);
-map.addLayer(driverLyr);
-
-pop = new nyc.ol.Popup(map);
-
-function mapClick(event){
-	var px = event.pixel, coords, zip, html = '';
-	map.forEachFeatureAtPixel(px, function(feature, layer){
-		var lyr = layer === riderLyr ? ' Riders' : ' Drivers';
-		zip = '<b>ZIP: ' + feature.getZip() + '</b><br>';
-		html += (feature.getCount() + lyr + '<br>');
-		coords = feature.getGeometry().getCoordinates();
-	});
-	if (coords){
-		$('#info').html(zip + html);
-		pop.setOffset([0, -10]);
-		pop.show({
-			coordinates: coords,
-			html: $('#info').show()
-		});
-	}
-};
-
-map.on('click', mapClick);
-
-new nyc.LocationMgr({
-	autoLocate: true,
-	controls: new nyc.ol.control.ZoomSearch(map),
-	locate: new nyc.ol.Locate(
-		new vote.Geocoder(GEOCODE_URL),
-		'EPSG:3857'
-	),
-	locator: new nyc.ol.Locator({
-		map: map,
-		layer: new ol.layer.Vector({source: new ol.source.Vector({})}),
-		zoom: 12
-	})
-});
-
-$('#ctl-z-srch input').attr({placeholder: 'Enter ZIP Code...', maxlength: 5});
-
-$(document).ready(function(){
-	$('#riders').change(function(){
-		riderLyr.setVisible(this.checked);
-	});
-	$('#drivers').change(function(){
-		driverLyr.setVisible(this.checked);
-	});
-});
+//get the data from API, using jQuery, assigns rider / driver variables to calls
+//make leaflet map, load custom icons, pop-ups, and add to a layer group
+$(document).ready(function () { 
+    $.when(
+    //wait until successful calls of both sources
+    $.getJSON('https://api.carpoolvote.com/live/unmatched-riders', function (riderSource) { 
+        jsonRider = riderSource;
```

Comment:
Good suggestion! You're right no need to be global here.  I've rewritten it here and changed `.then()` to `.done()`, commit here: https://github.com/voteamerica/voteamerica.github.io/pull/152/commits/f6fed7b949f80f52a7122d95ff059fdc7fa0fbad
```    $.when(
    //wait until successful calls of both sources
    $.getJSON('https://api.carpoolvote.com/live/unmatched-riders'),
    $.getJSON('https://api.carpoolvote.com/live/unmatched-drivers')
    
    ).done(function(riderSource, driverSource){
        var jsonRider = riderSource[0],
            jsonDriver = driverSource[0];```